lib/profileCache.js

```diff
@@ -11,41 +13,40 @@ const batch = require('./batch');
 // sets of scopes, but we have to limit it to a small set of known
 // possibilities in order to efficiently implement the "drop"
 // method below.
-const CACHEABLE_SCOPES = {
+
+const SCOPE_PROFILE = Scope.fromArray(['profile']);
+const SCOPE_UNTRUSTED = Scope.fromArray(['profile:display_name', 'profile:email', 'profile:uid']);
+
+const CACHEABLE_SCOPES = [
   // Trusted reliers all use the main "profile" scope.
-  'profile': true,
+  SCOPE_PROFILE.toJSON(),
   // Untrusted reliers tend to request these three scopes.
-  'display_name+email+uid': true
-};
-
+  SCOPE_UNTRUSTED.toJSON()
+];
 
-// Extract the profile-related scopes for a request.
-// This removes non-profile scopes and returns either
-// the string 'profile' or a sorted plus-separated list
-// of subscopes like 'display_name+email'.
 
-function extractProfileScopes(req) {
+function getProfileCacheKey(req) {
```

Comment:
The diff here got pretty awful, I I feel like the end result is not too bad.  It's slighly less "neat" than the previous version, but will do the same job and avoids having to parse a bunch of scopes here in this code.